server/mediapublic/mediapublic/views.py

```diff
@@ -594,8 +596,138 @@ def blogs_comment_by_id(request):
+        resp = {'error': 'Method Not Allowed'}
+        request.response.status = 405
+    return resp
+
+# [GET, POST]
+@view_config(route_name='user_playlists', renderer='json')
+def user_playlists(request):
+    cls = Playlists
+    resp = []
+    if request.method == "GET":
+        resp = [i.to_dict() for i in cls.get_by_user_id(person_id)]
+    elif request.method == "POST":
+        person_id = request.matchdict['uid']
+        keys = {}
+        playlist, status = do_post(request, keys, cls)
+        request.response.status = status
+        request.response.status = 303
+        # Which one should be used here? Maybe only redirect if == 200?
+        request.route_url('edit_user_playlist', uid=person_id, pid=playlist['id'])
+    else:
+        resp = {'error': 'Method Not Allowed'}
+        request.response.status = 405
+    return resp
+
+# [GET]
+@view_config(route_name='edit_user_playlist', renderer='json')
+def edit_user_playlist(request):
+    cls = Playlists
+    if request.method == "GET":
+        playlist_id = request.matchdict['id']
+        playlist = Playlists.get_by_id(playlist_id)
+        # Not using do_get_single because using sqlalchemy relationships from actual object, not dict
+        chosen_recordings = playlist.recordings
+        chosen_ids = [rec.id for rec in chosen_recordings]
+
+        unchosen_recordings = DBSession.query(
+            Recordings,
+        ).filter(
+            Recordings.id not in chosen_ids
+        )
+        # recordings user has in playlist vs recordings that can be added to it
+        # might want to limit unchosen, maybe take random 10
+
+        resp = {
+            'chosen_recordings': chosen_recordings,
+            'unchosen_recordings': unchosen_recordings
+        }
+    else:
+        resp = {'error': 'Method Not Allowed'}
+        request.response.status = 405
+    return resp
+
+
+# [GET]
+@view_config(route_name='new_user_playlist', renderer='json')
+def new_user_playlist(request):
+    # This should just return a form for title of new playlist.
+    resp = {}
+    if request.method == "GET":
+        resp = {'author_id': request.matchdict['uid']}
+        # pass through form
+    else:
+        resp = {'error': 'Method Not Allowed'}
+        request.response.status = 405
+    return resp
+
+# [           PUT        ]
+@view_config(route_name='assign_to_playlist')
+def assign_to_playlist(request):
+    cls = PlaylistAssignments
+    resp = {}
+    status = 500
+    if request.method == "PUT":
+        keys = {
+            'playlist_id': request.matchdict['id'],
+            'recording_id': request.body['recording_id'],
+            # allowed to use request.body? check_payload wouldn't work for this
+        }
+        resp, status = do_post(request, keys, cls)
+    else:
+        resp = {'error': 'Method Not Allowed'}
+        status = 405
+    request.response.status = status
+    return resp
+
+# [           PUT        ]
+@view_config(route_name='remove_from_playlist')
```

Comment:
This should use an implemented function called `delete_by_playlist_id and_recording_id(self, recording_id)`.  The end point will be `/users/:{uid}/playlists/:{pid}/unassign` ( or remove, whatever ), and the payload will have `recording_id` in it.  You then will get the playlist_id from `request.matchdict['pid']` and the `recording_id` from the payload, and then you can pass them into the `PlaylistAssignments.delete_by_playlist_id and_recording_id()` function where it will do the unassignment.  Additionally you could just call the function 'unassign()' and it takes in `palylist_id` and `recording_id` ... this might be a bit more elegant.